tracer/src/Datadog.Trace/Telemetry/TelemetryDataBuilder.cs

```diff
@@ -0,0 +1,109 @@
+﻿// <copyright file="TelemetryDataBuilder.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Telemetry
+{
+    internal class TelemetryDataBuilder
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<TelemetryDataBuilder>();
+        private int _sequence = 0;
+
+        public TelemetryData[] BuildTelemetryData(
+            ApplicationTelemetryData application,
+            HostTelemetryData host,
+            ConfigTelemetryData configuration,
+            ICollection<DependencyTelemetryData> dependencies,
+            ICollection<IntegrationTelemetryData> integrations,
+            bool sendHeartbeat)
+        {
+            if (application == null)
+            {
+                throw new ArgumentNullException(nameof(application));
+            }
+
+            if (configuration is not null)
+            {
+                Log.Debug("App initialized, sending app-started");
+                var payload = new AppStartedPayload
```

Comment:
Just to double-check, the configuration may be updated multiple times during an app's lifetime and cause this block to be entered multiple times. Does the backend accept multiple app-started events?

Reply:
Yes it does. Or at least, that was my understanding. But I'm going to double check just in case, thanks! 😄

Reply:
Confirmed, yes, this is fine. We can send subsequent payloads, and the `definitions` and `integrations` keys can be `null`, which matches my previous assumptions